runtime/src/identity.rs

```diff
@@ -108,6 +115,41 @@ pub enum LinkedKeyInfo {
     Group(Vec<IdentityId>),
 }
 
+pub type AuthorizationNonce = u64;
+
+/// It represents an authorization that any account could sing to allow operations related with a
+/// target identity.
+///
+/// # Safety
+///
+/// Please note, that `nonce` has been added to avoid **replay attack** and it should be the current
+/// value of nonce of master key of `target_id`. See `System::account_nonce`.
+/// In this way, the authorization is delimited to an specific transaction (usually the next one)
+/// of master key of target identity.
+#[derive(codec::Encode, codec::Decode, Clone, PartialEq, Eq, Debug)]
+pub struct TargetIdAuthorization<Moment> {
+    /// Target identity which is authorized to make an operation.
+    pub target_id: IdentityId,
+    /// It HAS TO be `target_id` authorization nonce: See `Identity::offchain_authorization_nonce`
+    pub nonce: AuthorizationNonce,
+    pub expires_at: Moment,
+}
+
+/// It is a signing item with authorization of that singning key (off-chain operation) to be added
+/// to an identity.
+/// `auth_signature` is the signature, generated by signing item, of `TargetIdAuthorization`.
+///
+/// # TODO
+///  - Replace `H512` type by a template type which represents explicitly the relation with
+///  `TargetIdAuthorization`.
+#[derive(codec::Encode, codec::Decode, Clone, PartialEq, Eq, Debug)]
+pub struct SigningItemWithAuth {
+    /// Signing item to be added.
+    pub signing_item: SigningItem,
+    /// Off-chain authorization signature.
+    pub auth_signature: H512,
```

Comment:
You can use the `OffChainSignature` type here instead of `H512`. Defined in the `utils.rs`

Reply:
@satyamakgec I am going to change that as part of [MESH-391](https://polymath.atlassian.net/browse/MESH-391)